prime-dotnet-webapi/Services/AccessTermService.cs

```diff
@@ -147,48 +147,37 @@ public async Task<bool> AccessTermExistsOnEnrolleeAsync(int accessTermId, int en
                 .AnyAsync();
         }
 
-        /// <summary>
-        /// Returns true if this access term has no newer versions
-        /// </summary>
-        public async Task<bool> IsCurrentAsync(int accessTermId)
-        {
-            // Currently, only the User Clause is versioned, and has different versions based on user class (OBO vs RU)
-            // Uses an anonymous object select to avoid fetching the user clause text, which we dont need.
-            // TODO: perhaps LINQ GroupBy to make this 1 DB call?
-
-            var userClause = await _context.AccessTerms
-                .Where(at => at.Id == accessTermId)
-                .Select(at => new
-                {
-                    EnrolleeClassification = at.UserClause.EnrolleeClassification,
-                    EffectiveDate = at.UserClause.EffectiveDate
-                })
-                .SingleAsync();
-
-            bool aNewerUserClause = await _context.UserClauses
-                .Where(uc => uc.EnrolleeClassification == userClause.EnrolleeClassification)
-                .AnyAsync(uc => uc.EffectiveDate > userClause.EffectiveDate);
-
-            return !aNewerUserClause;
-        }
-
         /// <summary>
         /// Returns true if the enrollees' most recent accepted access term has no newer versions
         /// </summary>
-        public async Task<bool> IsCurrentByEnrolleeAsync(int enrolleeId)
+        public async Task<bool> IsCurrentByEnrolleeAsync(Enrollee enrollee)
         {
-            var accessTermId = await _context.AccessTerms
-                .Where(at => at.EnrolleeId == enrolleeId)
+            var current = true;
+
+            var accessTerm = await _context.AccessTerms
+                .Include(at => at.AccessTermLicenseClassClauses)
+                .Where(at => at.EnrolleeId == enrollee.Id)
                 .Where(at => at.AcceptedDate != null)
                 .OrderByDescending(at => at.AcceptedDate)
-                .Select(at => at.Id)
                 .FirstOrDefaultAsync();
 
-            if (accessTermId == 0)
+            var currentAccessTerm = await GenerateAccessTermAsync(enrollee);
+
+            if (accessTerm.GlobalClauseId != currentAccessTerm.GlobalClause.Id
```

Comment:
I feel this chunk of code should probably be a method on AccessTerm, something like ClausesMatch()